stable/bundle_dep.pony

```diff
@@ -50,3 +52,88 @@ class BundleDepGitHub
     if git_tag isnt None then
       Shell("cd " + root_path() + " && git checkout " + (git_tag as String))
     end
+
+class BundleDepLocalGit
+  let bundle: Bundle
+  let info: JsonObject box
+  let package_name: String
+  let local_path: String
+  let git_tag: (String | None)
+  new create(b: Bundle, i: JsonObject box)? =>
+    bundle       = b
+    info         = i
+    local_path   = try info.data("local-path") as String
+                   else bundle.log("No 'local-path' key in dep: " + info.string()); error
+                   end
+    package_name = try _SubdirNameGenerator(local_path)
+                   else bundle.log("Something went wrong generating dir name "); error
+                   end
+    bundle.log(package_name)
+    git_tag      = try info.data("tag") as String
+                   else None
+                   end
+    bundle.log(package_name)
+
+  fun root_path(): String => ".deps/"+package_name
+  fun packages_path(): String => local_path
+
+  fun ref fetch()? =>
+    // TODO: Stop from doing the one shallow copy of .
+    Shell("rsync -avr --progress "+packages_path()+" "+root_path()+" --exclude .")
+    _checkout_tag()
+
+  fun _checkout_tag() ? =>
+    if git_tag isnt None then
+      Shell("cd " + root_path() + " && git checkout " + (git_tag as String))
+    end
+
+class BundleDepLocal
+  let bundle: Bundle
+  let info: JsonObject box
+  let package_name: String
+  let local_path: String
+  new create(b: Bundle, i: JsonObject box)? =>
+    bundle       = b
+    info         = i
+    local_path   = try info.data("local-path") as String
+                   else bundle.log("No 'local-path' key in dep: " + info.string()); error
+                   end
+    package_name = try _SubdirNameGenerator(local_path)
+                   else bundle.log("Something went wrong generating dir name "); error
+                   end
+
+  fun root_path(): String => ".deps/"+package_name
+  fun packages_path(): String => local_path
+
+  fun ref fetch()? =>
+    // TODO: Stop from doing the one shallow copy of .
+    Shell("rsync -avr --progress "+packages_path()+" "+root_path()+" --exclude .")
+
+primitive _SubdirNameGenerator
```

Comment:
I'd be open to any suggestions on how to improve this weird string processing code.